src/lib/chart/dendrogram.typ

```diff
@@ -99,33 +90,53 @@
 
   assert(mode in dendrogram-modes,
     message: "Invalid dendrogram mode. Use: " + repr(dendrogram-modes))
-  assert(type(x1-key) in (int, str))
-  assert(type(x2-key) in (int, str))
-  assert(type(height-key) in (int, str))
-
-  if size.at(0) == auto {
-    size.at(0) = (data.len() + 2)
+  assert(type(x1-key) in (int, str),
+    message: "Invalid x1-key type. Must be a integer or a string")
+  assert(type(x2-key) in (int, str),
+    message: "Invalid x2-key type. Must be a integer or a string")
+  assert(type(height-key) in (int, str),
+    message: "Invalid height-key type. Must be a integer or a string")
+
+  if ( size == auto ){
+    if mode == "vertical" {size = (auto, 1)} 
+    if mode == "horizontal" {size = (1, auto)}
```

Comment:
use an `if...else` statement here as mode can only be vertical or horizontal
```
if mode == "vertical" {
  ...
} else {
  ...
}
```

Reply:
Code will be needed when radial plots are implemented